include/Nodes1DProvisioner.hpp

```diff
@@ -69,44 +69,43 @@ namespace blitzdg {
       
-      const matrix_type & get_xGrid();
-      const vector_type & get_rGrid();
-      const matrix_type & get_Dr();
-      const matrix_type & get_V();
-      const matrix_type & get_J();
-      const matrix_type & get_rx();
-      const matrix_type & get_nx();
-
-      const index_vector_type & get_Fmask();
-      const matrix_type & get_Fx();
-      const matrix_type & get_Fscale();
-
-      const index_matrix_type & get_EToV();
-      const matrix_type & get_Lift();
+      const matrix_type & get_xGrid() const;
+      const vector_type & get_rGrid() const;
+      const matrix_type & get_Dr() const;
+      const matrix_type & get_V() const;
+      const matrix_type & get_J() const;
+      const matrix_type & get_rx() const;
+      const matrix_type & get_nx() const;
+
+      const index_vector_type & get_Fmask() const;
+      const matrix_type & get_Fx() const;
+      const matrix_type & get_Fscale() const;
+
+      const index_matrix_type & get_EToV() const;
+      const matrix_type & get_Lift() const;
     
-      const index_matrix_type & get_EToE();
-      const index_matrix_type & get_EToF();
+      const index_matrix_type & get_EToE() const;
+      const index_matrix_type & get_EToF() const;
 
-      const index_vector_type & get_vmapM();
-      const index_vector_type & get_vmapP();
+      const index_vector_type & get_vmapM() const;
+      const index_vector_type & get_vmapP() const;
 
-      const index_type get_mapI();
-      const index_type get_mapO();
-      const index_type get_vmapI();
-      const index_type get_vmapO();
+      index_type get_mapI() const;
+      index_type get_mapO() const;
+      index_type get_vmapI() const;
+      index_type get_vmapO() const;
 
-      index_type get_NumLocalPoints();
-      index_type get_NumElements();
-
-      // these can be moved to a helper (polynomials) class or made private within this class.
-      void computeJacobiPolynomial(vector_type const & x, const real_type alpha, const real_type beta, const index_type N, vector_type & p);
-      void computeJacobiQuadWeights(real_type alpha, real_type beta, index_type N, vector_type & x, vector_type & w);
-      void computeGaussLobottoPoints(real_type alpha, real_type beta, index_type N, vector_type & x);
-      void computeGradJacobi(vector_type const & x, const real_type alpha, const real_type beta, const index_type N, vector_type & dp);
+      index_type get_NumLocalPoints() const;
+      index_type get_NumElements() const;
 
+      void computeJacobiPolynomial(vector_type const & x, real_type alpha, real_type beta, index_type N, vector_type & p) const;
+      void computeJacobiQuadWeights(real_type alpha, real_type beta, index_type N, vector_type & x, vector_type & w) const;
+      void computeGaussLobottoPoints(real_type alpha, real_type beta, index_type N, vector_type & x) const;
+      void computeGradJacobi(vector_type const & x, real_type alpha, real_type beta, index_type N, vector_type & dp) const;
+      
```

Comment:
I've been thinking of moving this Jacobi guys to their own class, JacobiBuilders, or something like that, as nodes1DProvisioner is kind of bulky and the jacobi polynomial stuff is kind of its own set of work. What do you think about this, should we put this in the backlog?

Reply:
Sure, that sounds like a good idea to me. It would definitely make sense as its own standalone class.